docs/design/flow-traffic-mirroring/flow-traffic-mirroring.md

```diff
@@ -0,0 +1,172 @@
+      "receiver_point": {
+         "switch_id": string,
+         "port": int,
+         "inner_vlan": int,
+         "outer_vlan": int
+      }
+  }
+  ```
+
+* Delete mirror point:
+
+  `DELETE /flows/{flow_id}/mirror/{mirror_point_id}`
+
+
+* Get mirror points:
+
+  `GET /flows/{flow_id}/mirror`
+
+  Response payload:
+  ```
+  {
+      "flow_id": string,
+      "points":[
+          {
+              "mirror_point_id": string,
+              "mirror_direction": string [FORWARD|REVERSE],
+              "mirror_point_switch_id": string,
+              "receiver_point": {
+                  "switch_id": string,
+                  "port": int,
+                  "inner_vlan": int,
+                  "outer_vlan": int
+              }
+          }
+      ]
+  }
+  ```
+
+
+* API that needs to be updated: 
+  - Need to add information about the state of the mirror paths in the flow payload.
+  - It is necessary to add information about built mirror paths to API `GET /flows/{flow_id}/path`
+
+## Workflow
+
+Creation and deletion flow mirror points in the FlowHSTopology:
+
+![Create flow traffic mirror point](./create-mirror-point.png "Create flow traffic mirror point")
+![Delete flow traffic mirror point](./delete-mirror-point.png "Delete flow traffic mirror point")
+
+Getting flow mirror points in the NbWorkerTopology:
+
+![Get flow traffic mirror point](./get-mirror-point.png "Get flow traffic mirror point")
+
+## Affected kilda components
+* need to add mirror groups to the Floodlight;
+* need to update RerouteTopology to react to the network events for the mirror paths;
+* add logic to the flow update operation when updating flow endpoints;
+* update switch and flow validation.
+
+## Limitations
+It is allowed to use a transit switch as a mirror point only if flow is pinned.
+
+## Related issues
+There are currently no asymmetric bandwidth ISLs in the system. This feature sets paths in one direction only. 
+This results in asymmetric ISLs appearing in the system. This effect on the system requires more in-depth research.
+
+A possible solution to this issue is to create a dummy path that will consume bandwidth in the opposite direction.
+
+## Switch rules
+Examples of rules for VLAN transit encapsulation:
+* _Group_:
+```
+{
+   "dpid": dpid,
+   "type": "ALL",
+   "group_id": GROUP_ID,
+   "buckets":[
+      {
+         "actions":[
+            {
+               "type": "SET_FIELD", 
+               "field": "vlan_vid",
+               "value": vid
+            },
+            {
+               "type": "OUTPUT",
+               "port": first_out_port
+            }
+         ]
+      },
+      {
+         "actions":[
+            {
+               "type": "SET_FIELD", 
+               "field": "vlan_vid",
+               "value": vid
+            },
+            {
+               "type": "OUTPUT",
+               "port": second_out_port
+            }
+         ]
+      }
+   ]
+}
+```
+* _Mirror rule_ for egress rule:
```

Comment:
It should be better to highlight that for any kinf of exsiting OF flows (ingress, transit, eggress) existing actions set will be replaces with "goto group" action. And that group will have 2 buckets - first represents "usual" actions set, second represents mirror actions set (i.e. routint into mirror "path"). Because now you show 2 (transit it missing) mostly identical rules(where only matches not modigied by this feature are differ) without explanations.

Reply:
Description has been added.